src/nsrt_learning.py

```diff
@@ -91,7 +91,7 @@ def segment_trajectory(trajectory: GroundAtomTrajectory) -> List[Segment]:
         # like Holding, Handempty, etc., because when doing symbolic planning,
         # we only have predicates, and not the continuous parameters that would
         # be used to distinguish between a PickPlace that is a pick vs a place.
-        if traj.actions[t].has_option():
+        if not switch and traj.actions[t].has_option():
```

Comment:
this `if` can only make `switch` become True, so i think this minor optimization is ok